tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala

```diff
@@ -43,40 +45,72 @@ class ConfigurationPassImpl @Inject()(val options: PassOptions,
     * @return true, if the pass was successful
     */
   override def execute(): Boolean = {
-    var module = tlaModule.get
+    // try to read from the TLC configuration file
+    val configuredModule = tryConfigureFromTlcConfig(tlaModule.get)
 
+    // dump the configuration result
+    val outdir = options.getOrError("io", "outdir").asInstanceOf[Path]
+    PrettyWriter.write(configuredModule, new File(outdir.toFile, "out-config.tla"))
+
+    // set fallback options
+    setFallbackOptions()
+
+    // make sure that the operators are defined
+    ensureDeclarationsArePresent(configuredModule)
+
+    outputTlaModule = Some(configuredModule)
+    true
+  }
+
+  // if checker.init and checker.next are not set, set them to Init and Next, respectively
+  private def setFallbackOptions(): Unit = {
```

Comment:
Wouldn't it better to just set the `default` values of the CLI?

https://github.com/informalsystems/apalache/blob/4d6912e9a57a02bc28422fdd28924fa07788abb4/mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala#L27-L30

Reply:
True, but this does not work for the reasons in my earlier comment.